### ISSUE TITLE
Negative Loss

### ISSUE THREAD

**Issue by mmbejani** [JeanMaximilienCadic/ASRDeepSpeech#1]
I train the deep speech on Mozilla Persian Dataset (https://voice.mozilla.org/en/datasets). In the training procedure, always the value of the loss becomes negative then the parameters don't update. I read the document of the CTCLoss of pytorch that the output of the model should be logarithmized probabilities. Then, I add log_softmax function and the loss function becomes positive and this problem is solved, but another problem appears. The loss of the model stuck in a specific value, also the WRE and CRE don't change. What is the problem? (I can share my codes!)

**Comment by JeanMaximilienCadic** [comment]
It is likely that the problem comes from the preparation of your dataset. Please feel free to push to another branch you create. 
I will download the persian dataset and see the results I have.

**Comment by mmbejani** [comment]
Thanks a lot, I'll try on another language.

**Comment by mmbejani** [comment]
Thank you for responding. I can train DeepSpeech by adding log_softmax in the training phase at the last layer of the network. Besides, in the test phase, the last activation function is changed to softmax and all work well (after 200 epoch with batch_size 64, the WRE reaches 20%).
